### PR TITLE
The disgust system & food types

### PR DESCRIPTION
**Pull request by ghost**
Hello, I'm that washed up FTL coder, and someone asked me to make rats edible for lizards.

Now having recently learnt about defines & bitflags from Ninja. He gave me an idea in the good old way, memes: https://puu.sh/wMpAe/ee687b91eb.png

Now. A normal person would just do something simple and PR it. Not I. I decided I wanted more. I gave every food some food types. Now you might be wondering "the fuck do those do you tard"

THEY WORK WITH MY NEW DISGUST SYSTEM!

Now, in the current implementation, every species has a few disliked and toxic foods. Disliked foods simply make you go "ew". and raise your disgust a bit. Toxic foods raise it by a lot.

Now, being the good comrade I am, I turned rats into food. And gave them the foodtypes RAW | MEAT | GROSS. It then checks once you eat the food (in this case the rat) if your disliked or toxic foods are any of these. Where a normal human would be grossed out by raw or gross food. Lizards don't. because they're filth.

Once you're grossed out, you show symptoms of being ill for a while(not 100% happy with this yet) .

For now, this system is only used with pre-defined bitflags in the species datums. But the things you could do with this are unending.

Gluten intolerant mutation? Vegan mutation? fucking add it to the disliked_types list for someone and they'll be puking everything out as soon as they bite that non-vegan burger!

anyways the current foodtypes are as follows:

#define MEAT 1
#define VEGETABLES 2
#define RAW 4
#define JUNKFOOD 8
#define GRAIN 16
#define FRUIT 32
#define DAIRY 64
#define FRIED 128
#define ALCOHOL 256
#define GROSS 512
#define TOXIC 1024

its done and ready for merging, only thing that might still be changed is the effects of pukonium and the effects of being disgusted. I think a in-game test is best for this though.

:cl: Floyd (And sprites by Maya)
add: Disgust system - If you are disgusted with something you will enter a level of disgust for a while, currently only implemented with food & pukonium
add: Food types - Now each food has types based on their ingredients, and not all races can eat the same food. Humans cant eat raw and gross food, Lizards can't eat fruit.)
add: You can eat rats now, but it's mainly reccomended for lizards.
/:cl:

### CODE REVIEW

**Comment by Slinky5890** [comment]
This is what we get for me adding him to the coder chat. I take full responsibility for the meme PRs to come

**Comment by Jc1469** [comment]
this is a meme free zone. pr is denied. you can try and appeal this ban in a month

**Comment by ghost** [comment]
its not a meme if it's quality code

**Comment by Alexandria404** [comment]
I approve.

**Comment by ghost** [comment]
should i code a new reagent that makes people disgusted? then use it in something like a fart grenade. it would cause everyone in the area to start puking. thumbs up for yes thumbs down for no

**Comment by LarissaMayrink** [comment]
AW GOD, i love it, it has gone way beyond what i expected from that request.

**Comment by Draycu** [comment]
Oh man, this sounds amazing.  As a filthy lizard myself, I approve.

**Comment by ike709** [comment]
Floyd I will harm you if you don't add this to FTL13 when it's finished.

**Comment by ghost** [comment]
im gonna port it to TG if they want it so it'll be auto-mirrored

**Comment by LarissaMayrink** [comment]
@Qustinnus i belive that tg(much like hippie) already got the 'lizards can eat mice' thing, and given the pseudo freeze kor might not look good upon this.

**Comment by ghost** [comment]
We'll see. Also it's a refactor so that's fine, I can easily replace what they do.

**Comment by ghost** [comment]
I just checked and the freeze allows this because this is a refactor and doesn't affect syndicates

**Comment by ghost** [comment]
added pukeonium grenades.

File: code/modules/food_and_drinks/food/snacks.dm [hunk 1]
```diff
@@ -110,8 +109,17 @@
 			if(M.satiety > -200)
 				M.satiety -= junkiness
 			playsound(M.loc,'sound/items/eatfood.ogg', rand(10,50), 1)
+			var/fraction = min(bitesize / reagents.total_volume, 1)
+			if(ishuman(M))
+				var/mob/living/carbon/human/H = M
+				if(foodtype & H.dna.species.toxic_food)
+					M << fraction
+					M << "<span class='notice'>What the hell was that thing!</span>"
```
Comment (ike709) [inline code/modules/food_and_drinks/food/snacks.dm]: Either

>What the hell was that thing?

or

>What the hell was that thing?!

File: code/modules/food_and_drinks/food/snacks.dm [hunk 2]
```diff
@@ -110,8 +109,17 @@
...
+					M.adjust_disgust(25 + 30 * fraction)
+				else if(foodtype & H.dna.species.disliked_food)
+					M << "<span class='notice'>That didn't taste too good..</span>"
```
Comment (ike709) [inline code/modules/food_and_drinks/food/snacks.dm]: >That didn't taste very good...

File: code/modules/mob/living/carbon/human/examine.dm [hunk 3]
```diff
@@ -316,6 +316,13 @@
 		else
 			msg += "[t_He] [t_is] quite chubby.\n"
 
+	if(disgust >= DISGUST_LEVEL_DISGUSTED)
+		msg += "[t_He] looks disguted\n"
```
Comment (ike709) [inline code/modules/mob/living/carbon/human/examine.dm]: All three of these strings need periods.

Comment (ike709) [reply code/modules/mob/living/carbon/human/examine.dm]: Also, disguted is a typo.

File: code/modules/mob/living/carbon/human/life.dm [hunk 4]
```diff
@@ -397,6 +397,26 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
 			adjustToxLoss(4) //Let's be honest you shouldn't be alive by now
 
+	if(disgust)
+		var/pukeprob = 0.2 * disgust
+		if(disgust >= DISGUST_LEVEL_GROSS)
+			if(prob(25))
+				stuttering += 1
+				confused += 2
+			if(prob(10) && !stat)
+				src << "<span class='warning'>You feel kind of iffy...</span>"
```
Comment (ike709) [inline code/modules/mob/living/carbon/human/life.dm:407]: Vague sentence is vague. "Your stomach hurts..." or something?

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 5]
```diff
@@ -1473,3 +1473,15 @@ datum/reagent/romerol
 
 /datum/reagent/laughter/overdose_process(mob/living/M)
 	M.emote(pick(list("laugh","giggle")))
+
+
+/datum/reagent/pukonium
+	name = "pukonium"
+	id = "pukonium"
+	description = "Refined faraway in the lands of Adam El-Tablawania, this liquid is known to be the most vile thing on earth"
```
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: Period.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 6]
```diff
@@ -1473,3 +1473,15 @@ datum/reagent/romerol
...
+	color = "360f0f"
+
+/datum/reagent/pukonium/on_mob_life(mob/living/M)
+	M.adjust_disgust(2)
+	M << "<span class='notice'>OH GOD ITS SO VILE.</span>"
```
Comment (ike709) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: Exclamation point would probably be better.

Comment (ike709) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: Also the 'warning' span class or some other red one would probably be better.

File: code/modules/mob/living/carbon/human/examine.dm (hunk 3 above)
Comment (ghost) [reply code/modules/mob/living/carbon/human/examine.dm]: no his guts were dissed

File: code/modules/mob/living/carbon/human/life.dm (hunk 4 above)
Comment (ghost) [reply code/modules/mob/living/carbon/human/life.dm:407]: it's not just a system for being sick physically.

Comment (ghost) [reply code/modules/mob/living/carbon/human/life.dm:407]: you get a literal status alert if you're sick so if you miss it you're special

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 5 above)
Comment (ghost) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: you're on it

**Comment by ghost** [comment]
fixed grammar and pukonium nades for real this time

**Comment by GrayRachnid** [comment]
image

**Comment by ghost** [comment]
https://streamable.com/6qwh6 pukonium grenade

**Comment by GrayRachnid** [comment]
image

**Comment by ninjanomnom** [comment]
I regret teaching floyd bitflags, or anything really, now.

File: code/_onclick/hud/alert.dm [hunk 7]
```diff
@@ -141,13 +141,13 @@ The box in your backpack has an oxygen tank and gas mask in it."
...
-	name = "grossed out"
+	name = "Grossed out."
 	desc = "That was kind of gross..."
 	icon_state = "gross"
 
 /obj/screen/alert/verygross
-	name = "Very grossed out"
-	desc = "OH GOD I'M GOING TO VOMIT"
+	name = "Very grossed out."
+	desc = "I'm not feeling very well.."
```
Comment (ShadowDeath6) [inline code/_onclick/hud/alert.dm]: please use either three periods or one

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 8]
```diff
@@ -1473,3 +1473,15 @@ datum/reagent/romerol
 
 /datum/reagent/laughter/overdose_process(mob/living/M)
 	M.emote(pick(list("laugh","giggle")))
+
+
+/datum/reagent/pukonium
+	name = "pukonium"
+	id = "pukonium"
+	description = "Refined faraway in the lands of Adam El-Tablawania, this liquid is known to be the most vile thing on earth."
```
Comment (ShadowDeath6) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: player reference is kind of iffy

also capitalize Earth

Comment (matskuman5) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: if you want to make a fanneh mem reference make it Adamia or something so it'll at least have a chance of being "subtle"

File: code/_onclick/hud/alert.dm [hunk 9]
```diff
@@ -140,6 +140,21 @@ The box in your backpack has an oxygen tank and gas mask in it."
...
 	icon_state = "starving"
 
+/obj/screen/alert/gross
+	name = "Grossed out."
+	desc = "That was kind of gross..."
+	icon_state = "gross"
+
+/obj/screen/alert/verygross
+	name = "Very grossed out."
+	desc = "I'm not feeling very well.."
```
Comment (ShadowDeath6) [inline code/_onclick/hud/alert.dm:150]: one period or three please

Comment (ghost) [reply code/_onclick/hud/alert.dm:150]: Im only a poor period farmer I do not have any more periods to spare.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 8 above)
Comment (ghost) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: Im the opposite of subtle but sure someone write a new description and ill replace it

Comment (Mrcreepssss) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: what about "refined in the faraway lands of tablawnia, this liquid is known to be the most vile thing in the galaxy"

**Comment by Mrcreepssss** [comment]
so what does each species find gross?

File: code/_onclick/hud/alert.dm (hunk 9 above)
Comment (Mrcreepssss) [reply code/_onclick/hud/alert.dm:150]: then you can borrow some of mine ............................................................................................................ i'm rich, there's a 100% interest rate every day. ;)

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 8 above)
Comment (Cruix) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: Put pukonium in a separate PR

**Review by Cruix — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drinks/food/snacks_meat.dm [hunk 10]
```diff
@@ -212,25 +235,27 @@
...
 
 /obj/item/weapon/reagent_containers/food/snacks/nugget/New()
-    ..()
-    var/shape = pick(LUMP, STAR, LIZARD, CORGI)
-    desc = "A 'chicken' nugget vaguely shaped like a [shape]."
-    icon_state = "nugget_[shape]"
+	..()
+	var/shape = pick(LUMP, STAR, LIZARD, CORGI)
+	desc = "A 'chicken' nugget vaguely shaped like a [shape]."
+	icon_state = "nugget_[shape]"
```
Comment (Cruix) [inline code/modules/food_and_drinks/food/snacks_meat.dm:256]: Nice catch

**Comment by Cruix** [comment]
Also please put any sprites in the PR body
untitled

**Comment by ghost** [comment]
Sorry, I forgot :P

**Comment by ghost** [comment]
fixed everything + added liked foods

**Review by Cruix — APPROVED** [review]

**Pull request merged**